### PR TITLE
Clean concepts - models docs 

### PR DESCRIPTION
**Pull request by robertpiosik**
#### Description of what you did:

- remove capitalisation from file paths as models create lowercased files, eg. `restaurant.settings.json`, `restaurant.js`
- add `async` to code snippet in lifecycle callbacks
- rename middle parameter in `afterSave` hook
- add information how many hooks are there under user's disposal (10)
- to avoid confusion, model definition example should contain informations that are generated by default - it should include `options` property
- model definition file - `info.name` should be capitalised 
- simplify `attributes.name`, `attributes.description` by removing empty `default` fields. Change order of attributes to start with simpler types and end with relational
- component is, similarly to model - a database structure. Avoid confusion and stick with the same vocabulary. Change slightly description of a component model.
- add real world example to component model description
- remove dot from relative path in section "Where are the models defined?" making it an absolute path to project root
- move command in `For Content Types models` to the new line
- remove typo in word `bellow`
- modify example of a model settings to better show the purpose of custom definition (version 2 of the model)
- add description to `info` key descriprion
- rename User to restaurant in `Model options` to be consistent with other examples on the page

 ### Concern:
[Bookshelf example](https://strapi.io/documentation/3.0.0-beta.x/concepts/models.html#example-9) states that each callback receives `attrs` but in the actual autogenerated file it varies between `attrs` `response` and `columns`. Shouldn't we be more strict here?

### CODE REVIEW

**Comment by robertpiosik** [comment]
Follow up to this [comment](https://github.com/strapi/strapi/pull/6038#issuecomment-635512741)

File: docs/v3.x/concepts/models.md [hunk 1]
```diff
@@ -4,79 +4,86 @@
...
 module.exports = {
-  lifecycles: {
-    // Called before an entry is created
-    beforeCreate(data) {},
-    // Called after an entry is created
-    afterCreated(result) {},
-  },
+  // Before saving a value.
+  // Fired before an `insert` or `update` query.
+  beforeSave: async (model, attrs, options) => {},
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:15]: This isn't correct, they need to be inside of a lifecycles object.

File: docs/v3.x/concepts/models.md [hunk 2]
```diff
@@ -4,79 +4,86 @@
...
+In this example, there is a `Meta` component which contains the attributes `title` and `description` which are of type text. For better organization and easier attachment multiple components to a Content Type they are stored within categories. Above one lives in the `seo` category.
 
 ### Where are the models defined?
 
-For **Content Types**, models are defined in each `./api/**/models/` folder. Every JavaScript or JSON file in these folders will be loaded as a model. They are also available through the `strapi.models` and `strapi.api.**.models` global variables. Usable everywhere in the project, they contain the ORM model object that they refer to. By convention, a model's name should be written in lowercase.
+For **Content Types**, models are defined in each `/api/**/models/` folder. Every JavaScript or JSON file in these folders will be loaded as a model. They are also available through the `strapi.models` and `strapi.api.**.models` global variables. Usable everywhere in the project, they contain the ORM model object that they refer to. By convention, a model's name should be written in lowercase.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:86]: `/api...` is not valid when writing file paths either `./api` or `api/` but `/api` would mean that the the api folder is on the root of the file system (looking at it from a linux perspective)

File: docs/v3.x/concepts/models.md [hunk 3]
```diff
@@ -4,79 +4,86 @@
...
-In this example, there is a `Restaurant` model which contains the attributes `cover`, `name` and `description`.
+In this example, there is a `restaurant` model which contains the attributes `name`, `description` and `cover`.
 
 ### Component's models
 
-It also exist another type of models named `components`. A component is a data structure that can be used in one or many other API's model. There is no lifecycle related, only a JSON file definition
+A Component is a self contained database structure that can be reused in one or more Content Type. There is no lifecycle related, only a JSON file definition.
 
-**Path —** `./components/default/simple.json`
+**Path —** `/components/default/meta.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:61]: Again here either needs to be `./components/` or `components/`

File: docs/v3.x/concepts/models.md [hunk 4]
```diff
@@ -4,79 +4,86 @@
...
+  // After saving a value.
+  // Fired after an `insert` or `update` query.
+  afterSave: async (model, response, options) => {},
+
+  // ...and 10 more
 };
 ```
 
-**Path —** `./api/restaurant/models/Restaurant.settings.json`.
+**Path —** `/api/restaurant/models/restaurant.settings.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:25]: same pre-append the period or remove the prefix slash to the path.

File: docs/v3.x/concepts/models.md [hunk 5]
```diff
@@ -88,46 +95,77 @@ If you are just starting out it is very convenient to generate some models with
...
-- `globalId` (string) - Global variable name for this model (case-sensitive) - _only for Content Types_
-- `attributes` (object) - Define the data structure of your model. Find available options [bellow](#define-the-attributes).
+- `kind` (string): Define if the model is a Collection Type (`collectionType`) or a Single Type (`singleType`) - _only for Content Types_.
+- `connection` (string): Connection name which must be used. Default value: `default`.
+- `collectionName` (string): Collection name (or table name) in which the data should be stored.
+- `globalId` (string): Global variable name for this model (case-sensitive) - _only for Content Types_.
+- `attributes` (object): Define the data structure of your model. Find available options [below](#define-the-attributes).
 
-**Path —** `Restaurant.settings.json`.
+**Path —** `restaurant.settings.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:153]: Debating of it we should give the full path :thinking:

File: docs/v3.x/concepts/models.md [hunk 6]
```diff
@@ -164,7 +202,7 @@ The options key on the model-json states.
 
 - `timestamps`: This tells the model which attributes to use for timestamps. Accepts either `boolean` or `Array` of strings where first element is create date and second element is update date. Default value when set to `true` for Bookshelf is `["created_at", "updated_at"]` and for MongoDB is `["createdAt", "updatedAt"]`.
 
-**Path —** `User.settings.json`.
+**Path —** `restaurant.settings.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:205]: same here maybe full path?

File: docs/v3.x/concepts/models.md [hunk 7]
```diff
@@ -200,26 +238,26 @@ The following types are currently available:
...
 
 ### Exceptions
 
-- `uid` — This field type allows a `targetField` key. The value is the name of an attribute thas has `string` of `text` type.
+- `uid`: This field type allows a `targetField` key. The value is the name of an attribute thas has `string` of `text` type.
 
 ### Example
 
-**Path —** `Restaurant.settings.json`.
+**Path —** `restaurant.settings.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:260]: same here maybe full path?

File: docs/v3.x/concepts/models.md [hunk 8]
```diff
@@ -797,9 +835,9 @@ Dynamic Zone field let your create flexible space to content based on a componen
 
 #### Example
 
-Lets say we created an `slider` and `content` component in `article` category.
+Let's say we created a `slider` and a `content` component in an `article` category.
 
-**Path —** `./api/article/models/Article.settings.json`.
+**Path —** `/api/article/models/article.settings.json`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:840]: same needs a period or remove prefix slash to path

File: docs/v3.x/concepts/models.md [hunk 9]
```diff
@@ -999,7 +1037,7 @@ _Parameters:_
 
 ::: tab delete
 
-**`beforeDelete(params)`**
+**`beforeDeleted(params)`**
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:1040]: You are reverting a fix here, it shouldn't be `Deleted` it should be `Delete`

File: docs/v3.x/concepts/models.md [hunk 10]
```diff
@@ -1095,7 +1133,7 @@ _Parameters:_
 
 ### Example
 
-**Path —** `./api/user/models/User.js`.
+**Path —** `/api/user/models/user.js`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:1136]: same keep the period

File: docs/v3.x/concepts/models.md [hunk 11]
```diff
@@ -1149,7 +1187,7 @@ When you are building custom ORM specific queries the lifecycles will not be tri
 
 **Bookshelf example**
 
-**Path -** `./api/{apiName}/services/{serviceName}.js`
+**Path —** `/api/user/models/user.js`.
```
Comment (derrickmehaffy) [inline docs/v3.x/concepts/models.md:1190]: same keep the period

**Comment by derrickmehaffy** [comment]
@robertpiosik I would suggest you rebase this on the master as it would seem you are missing a few recently merged doc commits as I was seeing some regression to a few fixes. Likewise there is a lot of inconsistency in some of your changes.

**Comment by robertpiosik** [comment]
@derrickmehaffy I've made a silly thing to copy&paste changes from the pervious PR (which was targeting beta docs). 

I'll close this and submit a proper PR taking into account your suggestions.

**Comment by derrickmehaffy** [comment]
Thanks @robertpiosik :)

**Pull request closed without merging**